src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -3752,6 +3754,7 @@ impl ExplainStage {
             Self::GlobalPlan => Some(Global.path()),
             Self::PhysicalPlan => Some(Physical.path()),
             Self::Trace => None,
+            Self::PlanInsights => None,
```

Comment:
This is probably OK for an MVP, but might be impacting performance in regular `peek` mode when `emit_plan_insights_notice = true` because it will cause the `trace_plan(...)` calls in our optimizer to always create an entry. In other words, we will collect / clone every intermediate result in the optimizer pipeline in the accumulated `OptimizerTrace` instance, even though at the end we will extract only the `RawPlan` and `GlobalPlan` stages.

Reply:
Makes sense. How hard would it be to adjust the optimizer to allow ORing together multiple path filters?

Reply:
I added a commit to this PR that fixes the issue.